### PR TITLE
Consume method slot

### LINKED ISSUE #5159: add consume method slots to use-box in component header

### PR DESCRIPTION
**Pull request by odedre**
## Proposed Changes

- added `latest` to `component.gql` and `ComponentModel`, and using it in `component.ui.runtime` consume slots.
- use box gets its consume methods via slots registered to the component aspect via the `registerConsumeMethod` method
- methods currently registered from the component aspect.
- added pnpm consume method
- improved use box styles and added bottom drawer
- created a `expandable-tab-content` in `tab-content` component for ease of use for future methods.

closed #5159

### CODE REVIEW

File: scopes/component/component/component.docs.mdx [hunk 1]
```diff
@@ -6,3 +6,43 @@ labels: ['component', 'bit', 'extension', 'aspect']
...
+
+This aspect can be expanded via its available slots, to allow new capabilities to the component page ui.
+
+#### `registerConsumeMethod`:
+
+This method allows you to register new ways to consume a component, and present it in the component page ui. It will add any custom components to the `use-box` in the page's header.
+
+First, require the component aspect to your aspect's code, and use `registerConsumeMethod` to add your custom **consume method**.
+
+`registerConsumeMethod` expects a callback function that is provided with the `ComponentModel` data in it's `args` and returns an object with `component` and a `title` of type `ReactNode` that you would need to supply in order for them to render in the menu.
```
Comment (GiladShoham) [inline scopes/component/component/component.docs.mdx:24]: @odedre write something about that you can return undefined in title/component to not render it for specific conditions

Comment (GiladShoham) [reply scopes/component/component/component.docs.mdx:24]: also something about the ordering

File: scopes/component/component/ui/menu/menu.tsx [hunk 2]
```diff
@@ -66,29 +72,36 @@ function VersionRelatedDropdowns({ component }: { component: ComponentModel }) {
...
-  const packageVersion = useMemo(() => (isLatestVersion ? '' : `@${component.version}`), [component.version]);
-  const origin = typeof window !== undefined ? window.location.origin : undefined;
-  const finalElementsUrl = origin && component.elementsUrl ? `${origin}${component.elementsUrl}` : undefined;
-
+  // const isLatestVersion = useMemo(() => component.version === versionList[0], [component.version]);
+  // const packageVersion = useMemo(() => (isLatestVersion ? '' : `@${component.version}`), [component.version]);
+  // const origin = typeof window !== undefined ? window.location.origin : undefined;
+  // const finalElementsUrl = origin && component.elementsUrl ? `${origin}${component.elementsUrl}` : undefined;
+  const methods = getConsumeMethodsComponents(consumeMethods, component);
+  // console.log('method components', methods);
```
Comment (GiladShoham) [inline scopes/component/component/ui/menu/menu.tsx:80]: delete commented code?

File: scopes/component/component/ui/menu/menu.tsx [hunk 3]
```diff
@@ -66,29 +72,36 @@ function VersionRelatedDropdowns({ component }: { component: ComponentModel }) {
...
         />
       )}
       <VersionDropdown versions={versionList} currentVersion={component.version} />
     </>
   );
 }
+
+function getConsumeMethodsComponents(consumeMethods: OrderedConsumeSlot, componentModel: ComponentModel) {
+  const methods = flatten(consumeMethods.values());
+  // console.log('method', methods);
```
Comment (GiladShoham) [inline scopes/component/component/ui/menu/menu.tsx:97]: delete

File: scopes/component/component/ui/menu/menu.tsx [hunk 4]
```diff
@@ -66,29 +72,36 @@ function VersionRelatedDropdowns({ component }: { component: ComponentModel }) {
...
+  return useMemo(
+    () =>
+      methods.map((method) => {
+        // console.log('meth', method);
```
Comment (GiladShoham) [inline scopes/component/component/ui/menu/menu.tsx:101]: delete

File: scopes/component/component/ui/menu/nav-plugin.tsx [hunk 5]
```diff
@@ -1,9 +1,18 @@
...
   props: NavLinkProps;
   order?: number;
 };
 
 export type OrderedNavigationSlot = SlotRegistry<NavPlugin>;
+
+export type ConsumePlugin = (componentModel: ComponentModel) => {
+  Title?: ReactNode;
+  Component?: ReactNode;
+};
```
Comment (GiladShoham) [inline scopes/component/component/ui/menu/nav-plugin.tsx:16]: where is the order?

**Review by GiladShoham — CHANGES_REQUESTED** [review]

File: scopes/ui-foundation/use-box/dropdown/dropdown.tsx [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
 
 export type UseBoxDropdownProps = {
-  Menu: ComponentType;
+  Menu: React.ReactElement;
   containerClass?: string;
 } & Omit<DropdownProps, 'placeholder'>;
 
 export function UseBoxDropdown({ className, Menu, ...rest }: UseBoxDropdownProps) {
   const [key, setKey] = useState(0);
+  const NewMenu = React.cloneElement(Menu, { key });
```
Comment (odedre) [inline scopes/ui-foundation/use-box/dropdown/dropdown.tsx:14]: TODO: rename

File: scopes/component/component/component.ui.runtime.tsx [hunk 7]
```diff
@@ -7,12 +7,13 @@ import { NavLinkProps } from '@teambit/base-ui.routing.nav-link';
...
 import { isBrowser } from '@teambit/ui-foundation.ui.is-browser';
 import React from 'react';
+import { Install, Import } from '@teambit/ui-foundation.ui.use-box.menu';
 import { RouteProps } from 'react-router-dom';
 import CommandBarAspect, { CommandBarUI, CommandEntry } from '@teambit/command-bar';
 import copy from 'copy-to-clipboard';
 import { ComponentAspect } from './component.aspect';
 import { Component, ComponentPageElement, ComponentPageSlot } from './ui/component';
-import { Menu, NavPlugin, OrderedNavigationSlot } from './ui/menu';
+import { Menu, NavPlugin, OrderedNavigationSlot, OrderedConsumeSlot, ConsumePlugin } from './ui/menu';
```
Comment (KutnerUri) [inline scopes/component/component/component.ui.runtime.tsx:16]: "ordered" slot? 🤔

File: scopes/component/component/component.ui.runtime.tsx [hunk 8]
```diff
@@ -115,6 +118,91 @@ export class ComponentUI {
     },
   ];
 
+  private bitMethod: ConsumePlugin = (comp) => {
+    const version = this.getCurrentVersion(comp);
+    return {
+      Title: <img style={{ width: '20px' }} src="https://static.bit.dev/brands/bit-logo-text.svg" />,
```
Comment (KutnerUri) [inline scopes/component/component/component.ui.runtime.tsx:124]: classname? or better yet, `1em`/`1.25em`

File: scopes/component/component/component.ui.runtime.tsx [hunk 9]
```diff
@@ -115,6 +118,91 @@ export class ComponentUI {
...
+          componentId={`${comp.id.toString({ ignoreVersion: true })}${version}`}
+          packageName={`${comp.packageName}${version}`}
+          componentName={comp.id.name}
+        />
+      ),
+      order: 0,
+    };
+  };
+
+  // TODO - move to npm aspect
```
Comment (KutnerUri) [inline scopes/component/component/component.ui.runtime.tsx:136]: move

File: scopes/component/component/component.ui.runtime.tsx [hunk 10]
```diff
@@ -115,6 +118,91 @@ export class ComponentUI {
...
+          packageManager="npm"
+          copyString={`npm i ${comp.packageName}${packageVersion}`}
+          registryName={registry}
+        />
+      ),
+      order: 10,
+    };
+  };
+
+  // TODO - move to yarn? aspect
```
Comment (KutnerUri) [inline scopes/component/component/component.ui.runtime.tsx:155]: idunno? maybe move to npm aspect (or package manager aspect)

File: scopes/component/component/component.ui.runtime.tsx [hunk 11]
```diff
@@ -115,6 +118,91 @@ export class ComponentUI {
...
+  private yarnMethod: ConsumePlugin = (comp) => {
+    const registry = comp.packageName.split('/')[0];
+    const packageVersion = this.getCurrentVersion(comp);
+    return {
+      Title: (
+        <img style={{ height: '17px', paddingTop: '4px' }} src="https://static.bit.dev/brands/logo-yarn-text.svg" />
```
Comment (KutnerUri) [inline scopes/component/component/component.ui.runtime.tsx:161]: don't we have a bitcon for this? here and in the others

File: scopes/component/component/component.ui.runtime.tsx [hunk 12]
```diff
@@ -115,6 +118,91 @@ export class ComponentUI {
...
+          packageManager="pnpm"
+          copyString={`pnpm i ${comp.packageName}${packageVersion}`}
+          registryName={registry}
+        />
+      ),
+      order: 30,
+    };
+  };
+
+  getCurrentVersion(component: ComponentModel) {
```
Comment (KutnerUri) [inline scopes/component/component/component.ui.runtime.tsx:195]: I'd make it static (outside the class), or private

File: scopes/component/component/component.ui.runtime.tsx [hunk 13]
```diff
@@ -115,6 +118,91 @@ export class ComponentUI {
...
+  private yarnMethod: ConsumePlugin = (comp) => {
+    const registry = comp.packageName.split('/')[0];
+    const packageVersion = this.getCurrentVersion(comp);
+    return {
+      Title: (
+        <img style={{ height: '17px', paddingTop: '4px' }} src="https://static.bit.dev/brands/logo-yarn-text.svg" />
+      ),
+      Component: (
+        <Install
+          config={`npm config set ${registry}:registry' https://node.bit.dev`}
```
Comment (KutnerUri) [inline scopes/component/component/component.ui.runtime.tsx:165]: missing `'`? (here and in the others

File: scopes/component/component/component.ui.runtime.tsx [hunk 14]
```diff
@@ -115,6 +118,91 @@ export class ComponentUI {
...
+      order: 30,
+    };
+  };
+
+  getCurrentVersion(component: ComponentModel) {
+    const versionList = component.tags
+      ?.toArray()
+      .map((tag) => tag?.version?.version)
+      .filter((x) => x !== undefined)
+      .reverse();
```
Comment (KutnerUri) [inline scopes/component/component/component.ui.runtime.tsx:200]: can use or `.slice(-1).pop()` to the last item (assuming list is cloned)

File: scopes/component/component/component.ui.runtime.tsx [hunk 15]
```diff
@@ -147,7 +235,13 @@ export class ComponentUI {
 
   getMenu(host: string) {
```
Comment (KutnerUri) [inline scopes/component/component/component.ui.runtime.tsx:236]: better to make this a react component (good for hot reloading, etc)

File: scopes/component/component/component.ui.runtime.tsx [hunk 16]
```diff
@@ -183,30 +281,47 @@ export class ComponentUI {
...
+      pageSlot,
+      commandBarUI
+    );
     const section = new AspectSection();
 
     componentUI.commandBarUI.addCommand(...componentUI.keyBindings);
     componentUI.registerMenuItem(componentUI.menuItems);
     componentUI.registerRoute(section.route);
     componentUI.registerWidget(section.navigationLink, section.order);
+    componentUI.registerConsumeMethod(
```
Comment (KutnerUri) [inline scopes/component/component/component.ui.runtime.tsx:319]: cool

File: scopes/component/component/ui/menu/menu.tsx [hunk 17]
```diff
@@ -66,29 +72,28 @@ function VersionRelatedDropdowns({ component }: { component: ComponentModel }) {
...
-          )}
+          Menu={<ConsumeMethodsMenu methods={methods} componentName={component.id.name} />}
         />
       )}
       <VersionDropdown versions={versionList} currentVersion={component.version} />
     </>
   );
 }
+
+function getConsumeMethodsComponents(consumeMethods: OrderedConsumeSlot, componentModel: ComponentModel) {
```
Comment (KutnerUri) [inline scopes/component/component/ui/menu/menu.tsx:90]: I'd rename it - `useConsumeMethodSlot(consumeMethodsSlot, component)`
it's a hook, people might break the Rules Of Hooks, so it's important to call it `useXXX`

File: scopes/component/component/ui/menu/nav-plugin.tsx [hunk 18]
```diff
@@ -1,9 +1,19 @@
...
 
 export type OrderedNavigationSlot = SlotRegistry<NavPlugin>;
+
+export type ConsumePlugin = (componentModel: ComponentModel) => {
+  Title?: ReactNode;
+  Component?: ReactNode;
+  order?: number;
+};
+
+export type OrderedConsumeSlot = SlotRegistry<ConsumePlugin[]>;
```
Comment (KutnerUri) [inline scopes/component/component/ui/menu/nav-plugin.tsx:19]: I'd rename it - ConsumeMethodSlot (order is an implementation detail)

File: scopes/scope/scope/ui/menu/menu.tsx [hunk 19]
```diff
@@ -16,10 +16,6 @@ export function ScopeMenu({ className, ...rest }: MenuProps) {
 export function ScopeUseBox() {
   const scope = useContext(ScopeContext);
   return (
-    <UseBoxDropdown
-      position="bottom-end"
-      className={styles.useBox}
-      Menu={() => <ScopeUseBoxMenu scopeName={scope.name} />}
-    />
+    <UseBoxDropdown position="bottom-end" className={styles.useBox} Menu={<ScopeUseBoxMenu scopeName={scope.name} />} />
```
Comment (KutnerUri) [inline scopes/scope/scope/ui/menu/menu.tsx:19]: 👍

File: scopes/ui-foundation/use-box/dropdown/dropdown.tsx (hunk 6 above)
Comment (KutnerUri) [reply scopes/ui-foundation/use-box/dropdown/dropdown.tsx:14]: 👍

File: scopes/ui-foundation/use-box/menu/index.ts [hunk 20]
```diff
@@ -1,2 +1,11 @@
...
+export type { MenuProps, ConsumeMethod } from './menu';
+
+export { Import } from './import';
+export type { ImportProps } from './import';
+
+export { Install } from './install';
+export type { InstallProps } from './install';
+
+export { Registry } from './registry';
+export type { RegistryProps } from './registry';
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/index.ts:11]: I guess this works? It's like consume methods templates, what happens if a developer wants a different template?

File: scopes/ui-foundation/use-box/menu/install.tsx [hunk 21]
```diff
@@ -1,38 +1,39 @@
...
   registryName: string;
   packageManager: string;
-} & TabContentProps;
+} & ExpandableTabContentProps;
 
-export function Install({ componentName, copyString, back, registryName, packageManager, ...rest }: InstallProps) {
+export function Install({ componentName, copyString, registryName, packageManager, config, ...rest }: InstallProps) {
   return (
-    <TabContent
+    <ExpandableTabContent
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/install.tsx:20]: Isn't this our accordion? 🤔

File: scopes/ui-foundation/use-box/menu/menu.module.scss [hunk 22]
```diff
@@ -1,24 +1,22 @@
 .top {
-  padding: 15px 24px;
-  color: var(--bit-text-color, #414141);
+  padding: 16px 24px 24px 24px;
+  color: var(--bit-text-color-heavy, #2b2b2b);
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/menu.module.scss:3]: 👍

File: scopes/ui-foundation/use-box/menu/menu.module.scss [hunk 23]
```diff
@@ -1,24 +1,22 @@
...
   display: flex;
   align-items: center;
   font-size: var(--bit-p-xs);
   font-weight: bold;
-  margin-bottom: 12px;
   > span {
     &:first-child {
-      color: var(--bit-text-color-heavy, #2b2b2b);
-      font-weight: bold;
+      font-weight: inherit;
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/menu.module.scss:13]: check if needed

File: scopes/ui-foundation/use-box/menu/menu.tsx [hunk 24]
```diff
@@ -1,55 +1,42 @@
...
-import { Registry } from './registry';
-import { Install } from './install';
-import { Import } from './import';
+import { Tab } from '@teambit/ui-foundation.ui.use-box.tab';
+
 import styles from './menu.module.scss';
-import { Tabs } from './tabs';
-import { Elements } from './elements';
+
+// TODO where should we place this type? its being created in component ui runtime as well
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/menu.tsx:7]: The menu is the one actually using it, it makes sense it put it in the menu (and have the componentUi make a slot of it

File: scopes/ui-foundation/use-box/menu/menu.tsx [hunk 25]
```diff
@@ -1,55 +1,42 @@
...
-  const [activeRegistry, setActiveRegistry] = useState<string | undefined>(undefined);
-
-  if (activeRegistry === 'import') {
-    return <BitInfo prevTab={activeTab} setActive={() => setActiveRegistry(undefined)} />;
-  }
+export function Menu({ methods, componentName, ...rest }: MenuProps) {
+  const [activeTab, setActiveTab] = useState(0);
+  const content = useMemo(
+    () =>
+      methods.flat().sort((a, b) => {
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/menu.tsx:30]: we dont have polyfills for flat. use lodash.flatten, or maybe even lodash.orderBy

File: scopes/ui-foundation/use-box/menu/menu.tsx [hunk 26]
```diff
@@ -59,34 +46,18 @@ export function Menu({ packageName, componentId, registryName, componentName, el
...
-          registryName={registryName}
-          copyString={`yarn add ${packageName}`}
-          packageManager="yarn"
-          back={() => setActiveRegistry('install')}
-        />
-      )}
-      {activeTab === 'elements' && elementsUrl && <Elements componentName={componentName} url={elementsUrl} />}
+      <div className={styles.tabs}>
+        {content.map(({ Title, Component: comp }, index) => {
+          if (!Title || !comp) return null;
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/menu.tsx:51]: you said you will allow undefined for the whole content.. (by using `.filter(Boolean)` or `lodash.compact`)

File: scopes/ui-foundation/use-box/menu/menu.tsx [hunk 27]
```diff
@@ -1,55 +1,42 @@
...
-    return (
-      <Registry
-        prevTab={activeTab}
-        registryName={registryName}
-        copyString={`npm config set '${registryName}:registry' https://node.bit.dev`}
-        setActive={() => setActiveRegistry(undefined)}
-      />
-    );
-  }
+  const { Component } = content[activeTab] || {};
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/menu.tsx:39]: default value Component = null?

File: scopes/ui-foundation/use-box/menu/tooltip-copybox.tsx [hunk 28]
```diff
@@ -1,16 +1,16 @@
...
-};
+} & TooltipProps;
 
 // consider adding tooltip to the CopyBox directly, using `props.tooltip ? Tooltip : (Noop as Tooltip);`
 // forward ref required by tippy
-export function TooltipCopybox({ content }: TooltipCopyboxProps) {
+export function TooltipCopybox({ content, ...rest }: TooltipCopyboxProps) {
   return (
-    <Tooltip content={content} placement="bottom" maxWidth="" breakline={true}>
+    <Tooltip {...rest} content={content} disabled={content.length < 30} placement="bottom" maxWidth="" breakline={true}>
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/menu/tooltip-copybox.tsx:13]: cool

File: scopes/ui-foundation/use-box/menu/menu.docs.mdx [hunk 29]
```diff
@@ -5,12 +5,38 @@ labels: ['react', 'ui-component', 'component', 'menu']
 
 import { UseBoxDropdown } from '@teambit/ui-foundation.ui.use-box.dropdown';
 import { Menu } from './menu';
+import { Install } from './install';
+import { Import } from './import';
+
+Menu accepts an array of methods with a `Title` to be displayed in the tab and a `Component` to be rendered in the content section.
```
Comment (GiladShoham) [inline scopes/ui-foundation/use-box/menu/menu.docs.mdx:11]: add a sentence about order

File: scopes/ui-foundation/use-box/scope-menu/scope-menu.tsx [hunk 30]
```diff
@@ -28,19 +22,21 @@ export function Menu({ scopeName, ...rest }: MenuProps) {
           <Ellipsis>{`Bulk import from ${scopeName}`}</Ellipsis>
         </div>
       </div>
-      <TabContent
-        bottom={
-          <div className={linkStyles} onClick={() => setActive('import')}>
+      <ExpandableTabContent
+        content={
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/scope-menu/scope-menu.tsx:26]: one of these should be children

File: scopes/ui-foundation/use-box/tab-content/tab-content.module.scss [hunk 31]
```diff
@@ -9,3 +9,13 @@
...
 }
+
+.drawer {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+}
+
+.open {
+  transform: rotateZ(180deg);
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/tab-content/tab-content.module.scss:20]: hmm. this is repeating, we need a component for that

File: scopes/ui-foundation/use-box/menu/menu.tsx [hunk 32]
```diff
@@ -1,55 +1,42 @@
...
-export function Menu({ packageName, componentId, registryName, componentName, elementsUrl, ...rest }: MenuProps) {
-  const [activeTab, setActiveTab] = useState('bit');
-  const [activeRegistry, setActiveRegistry] = useState<string | undefined>(undefined);
-
-  if (activeRegistry === 'import') {
-    return <BitInfo prevTab={activeTab} setActive={() => setActiveRegistry(undefined)} />;
-  }
+export function Menu({ methods, componentName, ...rest }: MenuProps) {
+  const [activeTab, setActiveTab] = useState(0);
+  const content = useMemo(
```
Comment (GiladShoham) [inline scopes/ui-foundation/use-box/menu/menu.tsx:28]: not sure I would call it content. it's indeed the menu content, but later below, it is weird to have content.map

File: scopes/ui-foundation/use-box/menu/menu.tsx [hunk 33]
```diff
@@ -1,55 +1,42 @@
...
+    () =>
+      methods.flat().sort((a, b) => {
+        if (!a && !b) return 0;
+        if (a.order === undefined) return 1;
+        if (b.order === undefined) return -1;
+        return a?.order - b?.order;
+      }),
```
Comment (GiladShoham) [inline scopes/ui-foundation/use-box/menu/menu.tsx:35]: maybe use [lodash.orderBy](https://lodash.com/docs/4.17.15#orderBy)?

File: scopes/ui-foundation/use-box/dropdown/dropdown.composition.tsx [hunk 34]
```diff
@@ -38,15 +48,16 @@ UseBoxExample.canvas = {
...
-      <div style={styles}>
-        <UseBoxDropdown position="bottom-end" Menu={() => <ScopeMenu scopeName="@teambit.design" />}></UseBoxDropdown>
+      <div>
+        <UseBoxDropdown position="bottom-end" Menu={<ScopeMenu scopeName="@teambit.design" />}></UseBoxDropdown>
       </div>
     </ThemeCompositions>
   );
 };
 
 UseBoxScopeExample.canvas = {
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/dropdown/dropdown.composition.tsx:58]: canvas is for overview page only, better use style={} directly on the themeCompositions

**Review by GiladShoham — CHANGES_REQUESTED** [review]

File: scopes/ui-foundation/use-box/tab/tab.spec.tsx [hunk 35]
```diff
@@ -0,0 +1,31 @@
...
+  it('should render with the correct text', () => {
+    const { getByText } = render(<TabWithText />);
+    const rendered = getByText('bit');
+    expect(rendered).toBeTruthy();
+  });
+
+  it('should render without active class', () => {
+    const { getByText } = render(<TabWithText />);
+    const rendered = getByText('bit');
+    expect(rendered.classList.contains('active')).not.toBeTruthy();
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/tab/tab.spec.tsx:15]: might be able to do `expect(rendered.classList).not.toHave('active')`, or something similar (`not.toInclude('active')`)

File: scopes/ui-foundation/use-box/tab/tab.spec.tsx [hunk 36]
```diff
@@ -0,0 +1,31 @@
...
+    const rendered = getByAltText('bit-logo');
+    expect(rendered).toBeTruthy();
+  });
+});
+
+describe('active tab', () => {
+  it('should render with the active class', () => {
+    const { getByText } = render(<ActiveTab />);
+    const rendered = getByText('bit');
+    expect(rendered.classList.contains('active')).toBeTruthy();
```
Comment (KutnerUri) [inline scopes/ui-foundation/use-box/tab/tab.spec.tsx:29]: same

**Review by KutnerUri — APPROVED** [review]

File: scopes/ui-foundation/use-box/menu/index.ts (hunk 20 above)
Comment (odedre) [reply scopes/ui-foundation/use-box/menu/index.ts:11]: he can use the underlying component. this is more concrete

**Pull request merged**